File: packages/python/example/quickstart.py
"""Quickstart / screenshot demo — parse a bundled fixture, print the result.

No OCR needed; runs from anywhere in the repo.

    cd packages/python && python example/quickstart.py
"""

from __future__ import annotations

import json
import pathlib

from costco_price_tag_parser import OcrLine, parse_price_tag


def _repo_root() -> pathlib.Path:
    for p in pathlib.Path(__file__).resolve().parents:
        if (p / "spec" / "fixtures").is_dir():
            return p
    raise RuntimeError("repo root not found")


fixtures = json.loads((_repo_root() / "spec" / "fixtures" / "ocr_raw.json").read_text())

# A real Costco tag: NIKE synthetic gloves (item + price + unit price).
tag = parse_price_tag([OcrLine.from_json(e) for e in fixtures["IMG_3379.JPG"]])

out = tag.to_dict()
out.pop("rawText")  # drop the OCR blob for a clean view
print(json.dumps(out, ensure_ascii=False, indent=2))

File: packages/dart/example/quickstart.dart
// Quickstart / screenshot demo — parse a bundled fixture, print the result.
// No OCR needed; runs from anywhere in the repo.
//
//   cd packages/dart && dart run example/quickstart.dart
import 'dart:convert';
import 'dart:io';

import 'package:costco_price_tag_parser/costco_price_tag_parser.dart';

void main() {
  final fixtures = jsonDecode(
    File('${_repoRoot()}/spec/fixtures/ocr_raw.json').readAsStringSync(),
  ) as Map<String, dynamic>;

  // A real Costco tag: NIKE synthetic gloves (item + price + unit price).
  final lines = (fixtures['IMG_3379.JPG'] as List)
      .map((e) => OcrLine.fromJson(e as Map<String, dynamic>))
      .toList();

  final tag = parsePriceTag(lines);
  final out = tag.toJson()..remove('rawText'); // drop the OCR blob for a clean view
  stdout.writeln(const JsonEncoder.withIndent('  ').convert(out));
}

String _repoRoot() {
  var dir = Directory.current;
  while (true) {
    if (Directory('${dir.path}/spec/fixtures').existsSync()) return dir.path;
    final parent = dir.parent;
    if (parent.path == dir.path) throw StateError('repo root not found');
    dir = parent;
  }
}
